compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp

```diff
@@ -61,151 +64,13 @@ getSwizzledShape(ArrayRef<OpFoldResult> packedShape,
   return newShape;
 }
 
-static FailureOr<SmallVector<OpFoldResult>>
-getInnerTileSizesOfr(OpBuilder &rewriter, Location loc,
-                     RankedTensorType tensorType,
-                     const MaterializeEncodingInfo &materializeEncodingInfo,
-                     MaterializeEncodingValueFn materializeEncodingValueFn) {
-  ArrayRef<int64_t> staticTileSizes = materializeEncodingInfo.innerTileSizes;
-  if (llvm::all_of(staticTileSizes,
-                   [](int64_t i) { return !ShapedType::isDynamic(i); })) {
-    return getAsOpFoldResult(rewriter.getI64ArrayAttr(staticTileSizes));
-  }
-  assert(materializeEncodingValueFn &&
-         "When dynamic tile sizes are generated, a MaterializeEncodingValueFn "
-         "should be provided.");
-
-  FailureOr<MaterializeEncodingValueInfo> materializeEncodingValueInfo =
-      materializeEncodingValueFn(tensorType, rewriter, loc);
-  if (failed(materializeEncodingValueInfo)) {
-    return failure();
-  }
-  ArrayRef<Value> innerTileSizeValues =
-      materializeEncodingValueInfo->innerTileSizes;
-
-  SmallVector<OpFoldResult> result(staticTileSizes.size());
-  for (size_t i = 0; i < result.size(); ++i) {
-    if (ShapedType::isDynamic(staticTileSizes[i])) {
-      result[i] = innerTileSizeValues[i];
-    } else if (tensorType.isDynamicDim(i)) {
-      result[i] =
-          rewriter.create<arith::ConstantIndexOp>(loc, staticTileSizes[i])
-              .getResult();
-    } else {
-      result[i] = rewriter.getI64IntegerAttr(staticTileSizes[i]);
-    }
-  }
-  return result;
-}
-
-static void transposeInPlace(MaterializeEncodingInfo &info) {
```

Comment:
If a function was previously static to one .cpp file, why make it non-static now?

Reply:
Because it is used by other methods, e.g., see the below snippet. Furthermore, I think we will need it being public if we want to use the method for GPU data-tiling.

https://github.com/iree-org/iree/blob/7f9958760578bb84ecc61b1443690ce101b4867f/compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp#L202-L222

Reply:
> Because it is used by other methods, e.g., see the below snippet

oh, I see.

>  Furthermore, I think we will need it being public if we want to use the method for GPU data-tiling.

This is only used in transpose-narrow-N code, which we are not planning to use on GPU, for now. Things could always change, but a large part of the motivation for transpose-narrow-N on CPU is that ukernels are the "plan A" for codegen on CPU, while on GPU they are thought of as more of a temporary stop-gap until the LLVM AMDGPU back-end improves. Thus the emphasis on GPU remains on codegen, which shouldn't care much about the difference between narrow M and narrow N.